src/Common/src/System/Net/Logging/NetEventSource.Common.cs

```diff
@@ -0,0 +1,739 @@
+        [NonEvent]
+        public static void DumpArray(object thisOrContextObject, byte[] buffer, int offset, int count, [CallerMemberName] string memberName = null)
+        {
+            if (IsEnabled)
+            {
+                if (offset < 0 || offset > buffer.Length - count)
+                {
+                    Fail(thisOrContextObject, $"Invalid {nameof(DumpArray)} Args. Length={buffer.Length}, Offset={offset}, Count={count}", memberName);
+                    return;
+                }
+
+                count = Math.Min(count, MaxDumpSize);
+
+                byte[] slice = buffer;
+                if (offset != 0 || count != buffer.Length)
+                {
+                    slice = new byte[count];
+                    Buffer.BlockCopy(buffer, offset, slice, 0, count);
+                }
+
+                Log.DumpArray(IdOf(thisOrContextObject), memberName, slice);
+            }
+        }
+
+        /// <summary>Logs the contents of a buffer.</summary>
+        /// <param name="thisOrContextObject">`this`, or another object that serves to provide context for the operation.</param>
+        /// <param name="bufferPtr">The starting location of the buffer to be logged.</param>
+        /// <param name="count">The number of bytes to log.</param>
+        /// <param name="memberName">The calling member.</param>
+        [NonEvent]
+        public static unsafe void DumpArray(object thisOrContextObject, IntPtr bufferPtr, int count, [CallerMemberName] string memberName = null)
+        {
+            Debug.Assert(bufferPtr != IntPtr.Zero);
+            Debug.Assert(count >= 0);
+
+            if (IsEnabled)
+            {
+                var buffer = new byte[Math.Min(count, MaxDumpSize)];
+                fixed (byte* targetPtr = buffer)
+                {
+                    Buffer.MemoryCopy((byte*)bufferPtr, targetPtr, buffer.Length, buffer.Length);
+                }
+                Log.DumpArray(IdOf(thisOrContextObject), memberName, buffer);
+            }
+        }
+
+        [Event(DumpArrayEventId, Level = EventLevel.Verbose, Keywords = Keywords.Debug)]
+        private unsafe void DumpArray(string thisOrContextObject, string memberName, byte[] buffer) =>
+            WriteEvent(DumpArrayEventId, thisOrContextObject, memberName ?? MissingMember, buffer);
+        #endregion
+
+        #region Associate
+        /// <summary>Logs a relationship between two objects.</summary>
+        /// <param name="first">The first object.</param>
+        /// <param name="second">The second object.</param>
+        /// <param name="memberName">The calling member.</param>
+        [NonEvent]
+        public static void Associate(object first, object second, [CallerMemberName] string memberName = null)
+        {
+            DebugValidateArg(first);
+            DebugValidateArg(second);
+            if (IsEnabled) Log.Associate(IdOf(first), memberName, IdOf(first), IdOf(second));
```

Comment:
It's correct, or at least it's the behavior I intended.  The most common use case was just calling Associate(this, somethingElse), and I wanted to avoid the caller having to do Associate(this, this, somethingElse), so I added this overload that it does it for them.